### PR TITLE
[GSOC 2.1] Handle the division of Transfer Function instances.

### LINKED ISSUE #25308: How should division of Transfer Functions be handled
Phase 2 of my work in `GSOC` deals with improvements in `TransferFunction` methods and API. 

Right now division of Transfer Functions in SymPy leads to a ValueError.

```
>>> tf1 = TransferFunction.from_coeff_lists([1],[1,1],s)            # 1/(s+1)
>>> tf2 = TransferFunction.from_coeff_lists([2],[1,1],s)            # 2/(s+1)
>>> tf = tf1/tf2
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\kunni\sympy\sympy\sympy\physics\control\lti.py", line 929, in __truediv__
    raise ValueError("TransferFunction cannot be divided by {}.".
ValueError: TransferFunction cannot be divided by <class 'sympy.physics.control.lti.TransferFunction'>.
```
I went through previous discussions when `__truediv__` was implemented for the `TransferFunction` class, but did not find anything to justify it.

Relevant Comment :
1. https://github.com/sympy/sympy/pull/19390#discussion_r442357826

### PR DESCRIPTION
**Pull request by faze-geek**
#### References to other Issues or PRs

Fixes #25308 

#### Brief description of what is fixed or changed
As discussed division should be handled just like multiplication.

#### Release Notes

* physics.control
  * Modified the behavior of division of Transfer Functions.

### CODE REVIEW

**Review by Sc0rpi0n101 — APPROVED** [review]
Can you add that 1 test, please? Otherwise looks good to me.

File: sympy/physics/control/tests/test_lti.py [hunk 1]
```diff
@@ -962,7 +965,6 @@ def test_Feedback_functions():
     assert tf5 / (tf + tf5) == Feedback(tf5, tf)
 
     raises(TypeError, lambda: tf1*tf2*tf3 / (1 + tf1*tf2*tf3))
-    raises(ValueError, lambda: tf1*tf2*tf3 / tf3*tf5)
```
[the remark this answers is not in the document]
Comment (faze-geek) [reply sympy/physics/control/tests/test_lti.py:965]: Added the test back.

**Comment by faze-geek** [comment]
Not sure why this fails now. It seems that my change did not introduce this.
```
Run ruff check .
bin/sympy_time_cache.py:56:[13](https://github.com/sympy/sympy/actions/runs/5629040867/job/15253562866?pr=25329#step:8:14): F402 Import `time` from line 3 shadowed by loop variable
Found 1 error.
Error: Process completed with exit code 1.
```

**Review by Sc0rpi0n101 — APPROVED** [review]
not so sure about the forcing evaluation, but otherwise looks good.

File: sympy/physics/control/lti.py [hunk 2]
```diff
@@ -1218,7 +1223,13 @@ def __mul__(self, other):
         return Series(*arg_list, other)
 
     def __truediv__(self, other):
-        if (isinstance(other, Parallel) and len(other.args) == 2
+        if isinstance(other, TransferFunction):
+            return Series(*self.args, TransferFunction(other.den, other.num, other.var))
+        elif isinstance(other, Series):
+            tf_self = self.rewrite(TransferFunction)
+            tf_other = other.rewrite(TransferFunction)
+            return (tf_self) / (tf_other)
```
Comment (smichr) [inline sympy/physics/control/lti.py:1231]:
```suggestion
            return tf_self / tf_other
```

**Comment by faze-geek** [comment]
Thanks @smichr for the improvements !

**Pull request merged**